devices/Mpu6886/Mpu6886.cs

```diff
@@ -0,0 +1,371 @@
+        /// </summary>
+        /// <returns>Vector of the rotation</returns>
+        public Vector3 GetGyroscope() => GetRawGyroscope() * GyroscopeResolution;
+
+        /// <summary>
+        /// Reads the register of the on-chip temperature sensor which represents the MPU-6886 die temperature.
+        /// </summary>
+        /// <returns>Temperature in degrees Celcius</returns>
+        public Temperature GetInternalTemperature()
+        {
+            var rawInternalTemperature = GetRawInternalTemperature();
+
+            // p43 of datasheet describes the room temp. compensation calcuation
+            return new Temperature(rawInternalTemperature / 326.8 + 25.0, UnitsNet.Units.TemperatureUnit.DegreeCelsius);
+        }
+
+        private void WriteByte(Register register, byte data)
+        {
+            SpanByte buff = new byte[2]
+            {
+                (byte)register,
+                data
+            };
+
+            _i2c.Write(buff);
+        }
+
+        private short ReadInt16(Register register)
+        {
+            SpanByte val = new byte[2];
+            Read(register, val);
+            return BinaryPrimitives.ReadInt16LittleEndian(val);
+        }
+
+        private void Read(Register register, SpanByte buffer)
+        {
+            _i2c.WriteByte((byte)((byte)register));
+            _i2c.Read(buffer);
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            _i2c?.Dispose();
+            _i2c = null!;
+        }
+
+        /// <summary>
+        /// Calibrate the gyroscope by calculating the offset values and storing them in the GyroscopeOffsetAdjustment registers of the MPU6886.
+        /// </summary>
+        /// <param name="iterations">The number of sample gyroscope values to read</param>
+        /// <returns>The calulated offset vector</returns>
+        public Vector3 Calibrate(int iterations)
+        {
+            SetGyroscopeOffset(new Vector3(0, 0, 0));
+            Thread.Sleep(2);
+
+            var gyrSum = new double[3];
+
+            for (int i = 0; i < iterations; i++)
+            {
+                var gyr = GetRawGyroscope();
+
+                gyrSum[0] += gyr.X;
+                gyrSum[1] += gyr.Y;
+                gyrSum[2] += gyr.Z;
+
+                Thread.Sleep(2);
+            }
+
+            Vector3 offset = new Vector3(gyrSum[0] / iterations, gyrSum[1] / iterations, gyrSum[2] / iterations);
+
+            SetGyroscopeOffset(offset);
+
+            return offset;
+        }
+
+        /// <summary>
+        /// Write the gyroscope offset in the GyroscopeOffsetAdjustment registers of the MPU6886.
+        /// This function can be usefull when a custom callibration calculation is used, instead of the Calibrate function of this class.
+        /// </summary>
+        /// <param name="offset">The vector containing the offsets for the 3 axes.</param>
+        public void SetGyroscopeOffset(Vector3 offset)
```

Comment:
As you have a Get and a Set for this one and as it doesn't change over time, you can easily transform it as a property. Just rename those 2 functions as private and Create a GyroscopeOffet property calling those 2 private functions. This will be even nicer!